src/wallet/wallet.h

```diff
@@ -146,6 +151,94 @@ struct COutputEntry
     int vout;
 };
 
+/** An note outpoint */
+class JSOutPoint
+{
+public:
+    // Transaction hash
+    uint256 hash;
+    // Index into CTransaction.vjoinsplit
+    size_t js;
+    // Index into JSDescription fields of length ZC_NUM_JS_OUTPUTS
+    uint8_t n;
```

Comment:
I personally always find the ``char`` size rules confusing and cumbersome and I can't tell when my assumptions of "``char`` is 8 bits" or whether it's signed or not are correct. For all of these reasons I strongly prefer ``uint8_t`` which is unambiguous.